python/mleap/pyspark/feature/string_map.py

```diff
@@ -1,17 +1,37 @@
+import six
 from pyspark.ml.util import JavaMLReadable, JavaMLWritable, _jvm
 from pyspark.ml.wrapper import JavaTransformer
 from pyspark.ml.param.shared import HasInputCol, HasOutputCol
+from pyspark.sql import DataFrame
 
 
 class StringMap(JavaTransformer, HasInputCol, HasOutputCol, JavaMLReadable, JavaMLWritable):
     def __init__(self, labels, inputCol=None, outputCol=None, handleInvalid='error', defaultValue=0.0):
         """
         __init__(self, labels, inputCol=None, outputCol=None, handleInvalid='error', defaultValue=0.0)
-        labels is a dict {string: double}
+        labels must be a dict {string: double} or a spark DataFrame with columns inputCol & outputCol
         handleInvalid: how to handle missing labels: 'error' (throw an error), or 'keep' (map to the default value)
         """
-        assert handleInvalid in ['error', 'keep'], 'Invalid value for handleInvalid: {}'.format(handleInvalid)
         super(StringMap, self).__init__()
+
+        if isinstance(labels, DataFrame):
+            assert inputCol, 'inputCol is required when labels is a DataFrame'
+            assert outputCol, 'outputCol is required when labels is a DataFrame'
+            labels = {row[0]: float(row[1]) for row in labels.select([inputCol, outputCol]).collect()}
+
+        def validate_args():
+            """
+            validate args early to avoid failing at Py4j with some hard to interpret error message
+            """
+            assert handleInvalid in ['error', 'keep'], 'Invalid value for handleInvalid: {}'.format(handleInvalid)
```

Comment:
you could make handleInvalid be an Enum, rather than manually checking here

Reply:
Sorry, but what Enum? Something that works for all envs: py27,py36,py37 (https://github.com/combust/mleap/blob/master/python/tox.ini#L2)?

Reply:
I was thinking this one https://cpython-test-docs.readthedocs.io/en/latest/library/enum.html but it's more of a nice to have. We can also think about this again after 2020 when python 2.7 support is completely dropped (maybe eventually dropping it here too..).

Reply:
So, resolve this conversion maybe?